Tarea 6

H-Cote opened the pull request:
Se inicia la tarea 6

H-Cote committed:
Primera versión de la tarea 4.

Se inicia el ejercicio 1.

H-Cote committed:
Segunda versión de la tarea 4.

Se completan los ejercicos 1,2 y 3.
Nuevas versiones de AutomDiff.jl y runtest.jl

H-Cote committed:
Tercera versión de la tarea 4

Se da una nueva respuesta a la última pregunta.

H-Cote committed:
Actualización del repositorio
Merge branch 'master' of https://github.com/lbenet/2017-2_TSFisComputacional-1 into Tarea_4

H-Cote committed:
Primera versión de la tarea 5

Se inicia el ejercicio 1.

H-Cote committed:
Segunda versión de la tarea 5

Se avanza en el ejercicio 1.
Se dan una nuevas definiciones para las funciones.

H-Cote committed:
Actualización del repositorio

Merge branch 'master' of https://github.com/lbenet/2017-2_TSFisComputacional-1 into Tarea_5

H-Cote committed:
Primera versión de la tarea 6

Se inicia el ejercicio 2 con la implementación del método de Runge-Kutta

lbenet commented:
Lo primero que hay que decir, es que en este pull request hay cosas que corresponden a la tarea 5 (incompletas) y que **no** deberían estar aquí.

lbenet commented:
En cuanto a la Tarea 6, tomo esto como el primer envío. 

El Ejercicio 2 es sobre el problema $ dot{x} = x^2 $. Esto lo comento ya que me parece que estás resolviendo ahí el oscilador armónico, ¿no? Ignorando esto, las ecuaciones de movimiento (`g1` y `g2` las puede poner en una misma función, y no me parece necesario crear una función que te devuelve el `cos`, ya que `cos` directamente lo hace. Claramente hay un error en lo que has hecho.

H-Cote committed:
Actualización del repositorio
Merge branch 'master' of https://github.com/lbenet/2017-2_TSFisComputacional-1 into Tarea_6

H-Cote committed:
Segunda versión de la tarea 6

Se corrige el método de Runge-Kutta.
Se agregan los archivos taylor.jl y runtest_taylor.jl de la tarea 5
Se inicial el ejercicio 1. El uso del archivo runtest_taylor.jl tuvo errores, por lo que el código mostrado genera resultados dentro del notebook de la tarea 5

lbenet commented:
Creo que el error en los tests se debe (al menos parcialmente) a que tienes comentada la línea
```julia
#using Ty
```
por lo que no encuentra qué es `Taylor`. 

En cuanto a la implementación de Runge-Kutta, dado que usa un paso de integración `h` constante, entonces el tiempo final, después de N iteraciones, es $t_fin = t_0 + n h $, de donde puedes despejar n. Entonces, puedes dar como parámetros de tu función `rk4` a la función `f`, la condición inicial (x0), el tiempo inicial (t0), el final (tf) y el paso de integración (h).

H-Cote committed:
Tercera versión de la tarea 6

Se incluyen los archivos runtest_taylor.jl y taylor.jl de la sexta versión de la tarea 5
Se modifica la función integradora de Taylor, así como las funciones de Runge-Kutta de los ejercicios 2 y 3.

lbenet commented:
Algunos comentarios:

- El ejercicio sugiere que partas la función `integra_taylor` en tres funciones que hacen cosas más concretas. Sugiero que sigas la indicación. Otra cosa: no sólo te interesa que la función devuelva el valor de la variable dependiente, sino también el del tiempo!

- Me parece que en los integradores de Runge-Kutta, el ciclo `for` debe empezar en 1 (y no en cero). Eso hará que el tiempo final sea `tf` y no `tf+h`.
